python/ray/tests/test_state_api.py

```diff
@@ -2290,32 +2323,35 @@ class Actor:
         wait_for_condition(lambda: len(list_objects()) > 0)
 
         # Running 3 slow apis to exhaust the limits
-        res_q = mp.Queue()
-        start_q = mp.Queue()  # not used
+        res_q = queue.Queue()
+        start_q = queue.Queue()  # used for sync
         procs = [
-            mp.Process(
+            threading.Thread(
                 target=_try_state_query_expect_rate_limit,
                 args=(
                     list_workers,
                     res_q,
                     start_q,
                 ),
+                kwargs={"timeout": 3},
             ),
-            mp.Process(
+            threading.Thread(
                 target=_try_state_query_expect_rate_limit,
                 args=(
                     list_tasks,
                     res_q,
                     start_q,
                 ),
+                kwargs={"timeout": 3},
             ),
-            mp.Process(
```

Comment:
There seems to be issue with multiprocessing - I am trying to come up with a minimal reproducible script for this.